Orange/data/variable.py

```diff
@@ -585,16 +591,19 @@ class DiscreteVariable(Variable):
     def __init__(self, name="", values=(), ordered=False, compute_value=None,
                  *, sparse=False):
         """ Construct a discrete variable descriptor with the given values. """
-        self.values = list(values)
-        if not all(isinstance(value, str) for value in self.values):
+        values = list(values)  # some people (including me) pass a generator
+        if not all(isinstance(value, str) for value in values):
             raise TypeError("values of DiscreteVariables must be strings")
+
         super().__init__(name, compute_value, sparse=sparse)
+        self.values = values
```

Comment:
I'd rather make `self.values` a property and check `self._value_index - self._value` consistency in its setter. It's too easy to obtain inconsistent state with this api.

Reply:
True. But this commit is from #4422, on which *this* PR was based. #4422 was already merged with this code.

I agree that `values` should be a property and it should also be a tuple, not a list. But if I remember correctly, I already tried to change this once but ran into problems. At any rate, it's not in this PR, but if you'd like to try it, I support it.